src/scriptdev2/scripts/eastern_kingdoms/zulgurub/boss_venoxis.cpp

```diff
@@ -107,90 +144,97 @@ struct boss_venoxisAI : public ScriptedAI
-                {
                     if (Unit* pTempTarget = m_creature->GetMap()->GetUnit((*iter)->getUnitGuid()))
-                    {
                         if (pTempTarget->GetTypeId() == TYPEID_PLAYER && m_creature->CanReachWithMeleeAttack(pTempTarget))
                             ++uiTargetsInRange;
-                    }
-                }
 
                 // If there are more targets in melee range cast holy nova, else holy fire
                 // not sure which is the minimum targets for holy nova
-                if (uiTargetsInRange > 3)
+                if (uiTargetsInRange >= 3)
                     DoCastSpellIfCan(m_creature, SPELL_HOLY_NOVA);
-                else
-                {
-                    if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0))
-                        DoCastSpellIfCan(pTarget, SPELL_HOLY_FIRE);
-                }
 
-                m_uiHolySpellTimer = urand(4000, 8000);
+                m_uiHolyNovaTimer = urand(10000, 15000);
             }
             else
-                m_uiHolySpellTimer -= uiDiff;
+                m_uiHolyNovaTimer -= uiDiff;
+
+            if (m_uiHolyWrathTimer < uiDiff)
+            {
+                if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0))
+                    DoCastSpellIfCan(pTarget, SPELL_HOLY_WRATH);
+
+                m_uiHolyWrathTimer = urand(20000, 25000);
+            }
+            else
+                m_uiHolyWrathTimer -= uiDiff;
+
+            if (m_uiHolyFireTimer < uiDiff)
+            {
+                if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0))
+                    DoCastSpellIfCan(pTarget, SPELL_HOLY_FIRE);
+
+                m_uiHolyNovaTimer = urand(10000, 15000);
+            }
+            else
+                m_uiHolyFireTimer -= uiDiff;
 
             // Transform at 50% hp
             if (m_creature->GetHealthPercent() < 50.0f)
-            {
                 if (DoCastSpellIfCan(m_creature, SPELL_SNAKE_FORM, CAST_INTERRUPT_PREVIOUS) == CAST_OK)
                 {
                     DoCastSpellIfCan(m_creature, SPELL_PARASITIC_SERPENT, CAST_TRIGGERED);
                     DoScriptText(SAY_TRANSFORM, m_creature);
                     DoResetThreat();
                     m_bPhaseTwo = true;
                 }
-            }
         }
         // Snake phase
         else
         {
             if (m_uiPoisonCloudTimer < uiDiff)
             {
                 if (DoCastSpellIfCan(m_creature, SPELL_POISON_CLOUD) == CAST_OK)
-                    m_uiPoisonCloudTimer = 15000;
+                    m_uiPoisonCloudTimer = urand(15000, 20000);
             }
             else
                 m_uiPoisonCloudTimer -= uiDiff;
 
             if (m_uiVenomSpitTimer < uiDiff)
             {
                 if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0))
-                {
```

Comment:
Wow, would you look at that @xfurry, now this commit actually improves on the codestyle. you're welcome. Now would you please take this seriously?